sequence_filenames.py

```diff
@@ -126,6 +126,8 @@ def group_files_by_imageset(filenames):
     # filename itself.
     template = []
     for f in filenames:
+        if f and hasattr(f, "__fspath__"):
+            f = f.__fspath__()
```

Comment:
```suggestion
        if f:
            f = os.fspath(f)
```
Although, I'm not sure the reason for the `if f:` everywhere - is it sometimes empty in these routines?